go mod vendor fails with quic-go v0.21.0

Using v0.21.0 of quic-go, the following error occurs when executing the **go mod vendor** command
"github.com\lucas-clemente\quic-go@v0.21.0\internal\qtls\go118.go:3:1: expected 'package', found "quic-go doesn't build on Go 1.18 yet."
It works the same in both windows and linux environments.
Check out the "github.com\lucas-clemente\quic-go@v0.21.0\internal\qtls\go118.go" file with the following code
![image](https://user-images.githubusercontent.com/37330396/120570336-771dd580-c44a-11eb-970f-c1f389cd3d5a.png)
Is this some new way of writing?

Ah, commands `go mod vendor` will need to parse import statements from all Go files to tell what packages need to be vendored. This will ignore build tags by definition, because one wants to vendor all dependencies across all build configurations.

@marten-seemann I think you need to provide valid syntax, then, but pulling imports doesn't require typechecking, so you can still use a typechecking error rather than a syntax error:

```
$ cat f.go
package p

var _ int = "quic-go doesn't build on Go 1.18 yet."
$ go build f.go
# command-line-arguments
./f.go:3:5: cannot use "quic-go doesn't build on Go 1.18 yet." (type untyped string) as type int in assignment
$ go mod vendor
go: no dependencies to vendor
$ gofmt f.go
package p

var _ int = "quic-go doesn't build on Go 1.18 yet."
```

--- a/internal/qtls/go118.go
+++ b/internal/qtls/go118.go
@@ -1,3 +1,5 @@
 // +build go1.18
 
-"quic-go doesn't build on Go 1.18 yet."
+package qtls
+
+var _ int = "quic-go doesn't build on Go 1.18 yet."

